src/array/cpu/rowwise_sampling.cc

```diff
@@ -27,17 +27,41 @@ inline FloatArray DoubleSlice(FloatArray array, const IdxType* idx_data,
   return ret;
 }
 
-template <typename IdxType, typename FloatType>
+template <typename IdxType, typename DType>
+inline NumPicksFn<IdxType> GetSamplingNumPicksFn(
+    int64_t num_samples, FloatArray prob, bool replace) {
+  NumPicksFn<IdxType> num_picks_fn = [prob, num_samples, replace]
+    (IdxType rowid, IdxType off, IdxType len,
+     const IdxType* col, const IdxType* data) {
+      const int64_t max_num_picks = (num_samples == -1) ? len : num_samples;
+      const DType* prob_data = prob.Ptr<DType>();
+      IdxType nnz = 0;
+      for (IdxType i = off; i < off + len; ++i) {
```

Comment:
just curious, does  `i` of `IdxType` handle correctly if `prob.length() > maximum IdxType`? For example, int32_t for row/col, but num_edges has to be int64_t.
should we always use `int64_t` for indexing?

Reply:
Good question.  For now I think if `IdxType` is `int32`, we assume that the number of edges is also less than `2 ** 31`, because otherwise the edge IDs in CSR will overflow.  That being said, I don't know if we perform any check on this.